### PR TITLE
Enabling the MetaMask Security Code Scanner

### PR DESCRIPTION
**Pull request by witmicko**
## Required Action

Prior to merging this pull request, please ensure the following has been completed:
- [x] The lines specifying `branches` correctly specifies this repository's default branch (usually `main`).
- [ ] Any paths you would like to ignore have been added to the `paths_ignored` configuration option (see [setup](https://github.com/MetaMask/Security-Code-Scanner/blob/main/README.md#setup))
- [x] Any existing CodeQL configuration has been disabled.

## What is the Security Code Scanner?
This pull request enables the [MetaMask Security Code Scanner](https://github.com/metamask/Security-Code-Scanner/) GitHub Action. This action runs on each pull request, and will flag potential vulnerabilities as a review comment. It will also scan this repository's default branch, and log any findings in this repository's [Code Scanning Alerts Tab](https://github.com/metamask/snaps/security/code-scanning). 

Screenshot 2024-02-12 at 9 19 05 PM

The action itself runs various static analysis engines behind the scenes. Currently, it is only running GitHub's CodeQL engine. For this reason, we recommend disabling any existing CodeQL configuration your repository may have.

## How do I interact with the tool?

Every finding raised by the Security Code Scanner will present context behind the potential vulnerability identified, and allow the developer to fix, or dismiss it.

The finding will automatically be dismissed by pushing a commit that fixes the identified issue, or by manually dismissing the alert using the button in GitHub's UI. If dismissing an alert manually, please add any additional context surrounding the reason for dismissal, as this informs our decision to disable, or improve any poor performing rules. 

 Screenshot 2024-02-12 at 8 41 46 PM

For more configuration options, please review the tool's [README](https://github.com/MetaMask/Security-Code-Scanner/blob/main/README.md). 
For any additional questions, please reach out to `@mm-application-security` slack.

### CODE REVIEW

File: .github/workflows/security-code-scanner.yml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        uses: MetaMask/Security-Code-Scanner@main
+        with:
+          repo: ${{ github.repository }}
+          paths_ignored: |
+            tests/
+            '**/test/'
+            '**/__mocks__/'
+            '**/__snapshots__/'
+            '**/*.test.js*'
+            '**/*.test.ts*'
```
Comment (FrederikBolding) [inline .github/workflows/security-code-scanner.yml:27]:
```suggestion
            '**/*.test.ts*'
            '**/*.test.browser.ts*'
```

File: .github/workflows/security-code-scanner.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+      contents: read
+      security-events: write
+    steps:
+      - name: MetaMask Security Code Scanner
+        uses: MetaMask/Security-Code-Scanner@main
+        with:
+          repo: ${{ github.repository }}
+          paths_ignored: |
+            tests/
+            '**/test/'
```
Comment (FrederikBolding) [inline .github/workflows/security-code-scanner.yml:23]:
```suggestion
            '**/test/'
            '**/test-utils/'
```

File: .github/workflows/security-code-scanner.yml [hunk 3]
```diff
@@ -0,0 +1,33 @@
+name: 'MetaMask Security Code Scanner'
+
+on:
+  push:
+    branches: ['main']
+  pull_request:
+    branches: ['main']
```
Comment (FrederikBolding) [inline .github/workflows/security-code-scanner.yml:7]:
```suggestion
```

Is there a reason to not just run this on every PR?

File: .github/workflows/security-code-scanner.yml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+      - name: MetaMask Security Code Scanner
+        uses: MetaMask/Security-Code-Scanner@main
+        with:
+          repo: ${{ github.repository }}
+          paths_ignored: |
+            tests/
+            '**/test/'
+            '**/test-utils/'
+            '**/__mocks__/'
+            '**/__snapshots__/'
```
Comment (Mrtenz) [inline .github/workflows/security-code-scanner.yml:26]:
```suggestion
            '**/__snapshots__/'
            '**/__fixtures__/'
```

File: .github/workflows/security-code-scanner.yml (hunk 3 above)
Comment (FrederikBolding) [reply .github/workflows/security-code-scanner.yml:7]: Not often, but we do sometimes have PRs that target other branches that then go into main later.

I just don't see a reason to limit where this is run!

**Review by Mrtenz — APPROVED** [review]

**Pull request merged**